XenAdmin/Program.cs

```diff
@@ -990,9 +998,59 @@ public static void ReconfigureConnectionSettings()
                 Properties.Settings.Default.ProvideProxyAuthentication = false;
                 Properties.Settings.Default.Save();
             }
+            else
+            {
+                ReconfigureProxyAuthenticationSettings();
+            }
             XenAPI.Session.Proxy = XenAdminConfigManager.Provider.GetProxyFromSettings(null);
         }
 
+        /// <summary>
+        /// Stores the Basic and Digest authentication modules, used for proxy server authentication, 
+        /// for later use; this is needed because we cannot create new instances of them and it 
+        /// saves us needing to create our own custom authentication modules.
+        /// </summary>
+        private static void RememberProxyAuthenticationModules()
+        {
+            var authModules = AuthenticationManager.RegisteredModules;
+            while (authModules.MoveNext())
+            {
+                var module = (IAuthenticationModule)authModules.Current;
+                if (module.AuthenticationType == "Basic")
+                    BasicAuthenticationModule = module;
+                else if (module.AuthenticationType == "Digest")
+                    DigestAuthenticationModule = module;
+            }
+        }
+
+        /// <summary>
+        /// Configures .NET's AuthenticationManager to only use the authentication module that is 
+        /// specified in the ProxyAuthenticationMethod setting. Also sets XenAPI's HTTP class to 
+        /// use the same authentication method.
+        /// </summary>
+        private static void ReconfigureProxyAuthenticationSettings()
+        {
+            var authModules = AuthenticationManager.RegisteredModules;
+            var modulesToUnregister = new List<IAuthenticationModule>();
+
+            while (authModules.MoveNext())
+            {
+                var module = (IAuthenticationModule)authModules.Current;
+                modulesToUnregister.Add(module);
+            }
+
+            foreach (var module in modulesToUnregister)
+                AuthenticationManager.Unregister(module);
+
+            var authSetting = (HTTPHelper.ProxyAuthenticationMethod)Properties.Settings.Default.ProxyAuthenticationMethod;

```

Comment:
If the value is manually changed to 2 then authSetting gets set to 2 and the 'else' clause below it will set the .NET authentication manager to use Digest, so at the moment any values that are not 1 sets XenCenter to use Digest. This matches up with the radio button display behaviour, which checks the Digest radio button for invalid values. We could either:
1. Leave it like this so they both match, which I prefer.
2. Have an if/elseif here which, if the setting is manually made invalid, will make the authentication manager not able to use any authentication method (which will disallow any connection that needs authentication), but then the setting in the connection options page will still display Digest as default.